strawberry/custom_scalar.py

```diff
@@ -0,0 +1,69 @@
+from graphql.type.scalars import GraphQLScalarType
+
+from .type_converter import REGISTRY
+
+
+def _process_scalar(cls, *, description, serialize, parse_value, parse_literal):
+    name = cls.__name__
+
+    REGISTRY[cls] = GraphQLScalarType(
+        name=name,
+        description=description,
+        serialize=serialize,
+        parse_value=parse_value,
+        parse_literal=parse_literal,
+    )
+
+    return cls
+
+
+def scalar(
+    cls=None, *, description=None, serialize=None, parse_value=None, parse_literal=None
+):
+    """Annotates a class or type as a GraphQL custom scalar.
+
+    Example usages:
+
+    >>> strawberry.scalar(
+    >>>     datetime.date,
+    >>>     serialize=lambda value: value.isoformat(),
+    >>>     parse_value=datetime.parse_date
+    >>> )
+
+    >>> Base64Encoded = strawberry.scalar(
+    >>>     NewType("Base64Encoded", bytes),
+    >>>     serialize=base64.b64encode,
+    >>>     parse_value=base64.b64decode
+    >>> )
+
+    >>> @strawberry.scalar(
+    >>>     serialize=lambda value: ",".join(value.items),
+    >>>     parse_value=lambda value: CustomList(value.split(","))
+    >>> )
+    >>> class CustomList:
+    >>>     def __init__(self, items):
+    >>>         self.items = items
+
+    """
+
+    if serialize is None:
+
+        def serialize(x):
+            return x
```

Comment:
do we have to create a function everytime? we could use a generic one, no?

Reply:
Yes, I changed it.